src/asset/asset-list-loader.js

```diff
@@ -0,0 +1,215 @@
+import { EventHandler } from '../core/event-handler.js';
+
+import { Asset } from './asset.js';
+
+/**
+ * Used to load a group of assets and fires a callback when all assets are loaded.
+ *
+ * @augments EventHandler
+ */
+class AssetListLoader extends EventHandler {
+    /**
+     * Create a new AssetListLoader using a list of assets to load and the asset registry used to load and manage them.
+     *
+     * @param {Asset[]|number[]} assetList - An array of {@link Asset} objects to load or an array of Asset IDs to load.
+     * @param {AssetRegistry} assetRegistry - The application's asset registry.
+     * @example
+     * const assetListLoader = new pc.AssetListLoader([
+     *     new pc.Asset("texture1", "texture", { url: 'http://example.com/my/assets/here/texture1.png') }),
+     *     new pc.Asset("texture2", "texture", { url: 'http://example.com/my/assets/here/texture2.png') })
+     * ], pc.app.assets);
+     */
+    constructor(assetList, assetRegistry) {
+        super();
+        this._assets = [];
+        this._loadingAssets = new Set();
+        this._registry = assetRegistry;
+        this._loaded = false;
+        this._count = 0; // running count of successfully loaded assets
```

Comment:
I think we could remove _count, and just simply remove assets when done (successfully or not) from _loadingAssets. When the set is empty, we're done.